spring-context/src/test/java/org/springframework/context/annotation/configuration/AutowiredConfigurationTests.java

```diff
@@ -506,4 +516,8 @@ public TestBean testBean() throws IOException {
 		}
 	}
 
+
+	@Component
+	static record RecordBean(@Value("recordBeanName") String name) {
```

Comment:
To access the `recordBeanName` system property, this would need to be changed to `@Value("${recordBeanName}")` or `@Value("#{systemProperties[recordBeanName]}")`.

I will address this after merging this PR, so there is no need to modify this PR.